src/main/scala/com/fulcrumgenomics/vcf/MakeMixtureVcf.scala

```diff
@@ -0,0 +1,268 @@
+    * all specified samples exist in the VCF.
+    *
+    * @param header the input VCF header
+    * @return the set of samples with proportions
+    */
+  private[vcf] def determineSamples(header: VCFHeader): Seq[Sample] = this.samples match {
+    case Seq() =>
+      val ss       = header.getSampleNamesInOrder
+      val fraction = 1 / ss.size().toDouble
+      header.getSampleNamesInOrder.map(s => Sample(s, fraction)).toSeq
+    case sampleStrings =>
+      // Extract the samples and fractions from their strings
+      var samples = sampleStrings.map { s =>
+        s.lastIndexOf('@') match {
+          case -1 => Sample(s)
+          case  i => Sample(s.substring(0, i), s.substring(i+1).toDouble)
+        }
+      }
+
+      // Validate that each sample was specified only once
+      validate(samples.size == samples.map(_.name).distinct.size, "Each sample name can be specified at most once.")
+
+      // Auto-calculate fraction for any samples that didn't have it
+      val total = samples.map(_.proportion).sum
+      validate(total <= 1, "Sample fractions added to more than 1.0")
+      val remainder = 1 - total
+      val fraction  = remainder / samples.count(_.proportion == 0)
+      samples = samples.map(s => if (s.proportion == 0) s.copy(proportion = fraction) else s)
+      validate(samples.map(_.proportion).sum == 1, "Sample fractions must add to 1.0")
+
+      // Then lastly validate that all given samples exist in the VCF
+      val missing = samples.filterNot(s => header.getSampleNamesInOrder.contains(s.name))
+      validate(missing.isEmpty, s"Samples are not present in the VCF: ${missing.mkString(", ")}")
+
+      samples
+  }
+
+  /** The main execute method that reads the inputs, mixes and writes the outputs. */
+  override def execute(): Unit = {
+    val in            = new VCFFileReader(input.toFile, false)
+    val samples       = determineSamples(in.getFileHeader)
+    val sampleNameSet = CollectionUtil.makeSet(samples.map(_.name):_*)
+    val out           = MakeMixtureVcf.makeWriter(output, in.getFileHeader, MakeMixtureVcf.HeaderLines, this.outputSampleName)
+    val fmt           = new DecimalFormat("0." + ("#" * precision))
+
+    logger.info("Making mixture of: ", samples.mkString(", "))
+
+    in.map(_.subContextFromSamples(sampleNameSet)).foreach { ctx =>
+      val outputGt = {
+        if (!this.noCallIsHomRef && ctx.getNoCallCount > 0) {
+          MakeMixtureVcf.noCall(this.outputSampleName)
+        }
+        else {
+          if (ctx.getStart == 11288758) {
+            logger.info("Hello!")
+          }
+          val alleleFractions = mutable.Map[Allele, Double](ctx.getAlleles.toSeq.map(a => a -> 0d):_*)
+          samples.foreach(updateAlleleFractionsForSample(ctx, _, alleleFractions))
+
+          // This horror (turning the AF into a string instead of a double[]) brought to you by the
+          // fact that htsjdk's VCF code will otherwise format the doubles by rounding to 3 decimal places!
```

Comment:
Starting a discussion on this here: https://github.com/samtools/htsjdk/issues/793